### PR TITLE
mailroom part 2

### PR DESCRIPTION
**Pull request by ikempcode33**
Ready for grading.

### CODE REVIEW

File: students/Isabella_Kemp/lesson04/mailroom_part2.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                donors[name] = record
+                email(name, donation)
+
+            else:
+                # new donor is not in our list, add the donor to our list
+                # get donation amount, add donation to list, send thank you
+                print("!!! NEW DONOR !!!")
+                donation = get_donation()
+                donors.update({name: (donation)})
+                email(name, donation)
```
Comment (charlesrobison) [inline students/Isabella_Kemp/lesson04/mailroom_part2.py]: This isn't allowing me to exit once new donors/donations are added.

File: students/Isabella_Kemp/lesson04/mailroom_part2.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+# end thank_you
+
+
+# Checks to see if provided name is in our donor list
+# returns donor record if found, else None
+def get_donor(donor_name):
+    try:
+        return donors[donor_name]
+    except:
```
Comment (charlesrobison) [inline students/Isabella_Kemp/lesson04/mailroom_part2.py]: Generally want to avoid an empty except.

**Comment by charlesrobison** [comment]
@natasha-aleksandrova - note there's an update to gitignore to the root on this PR

**Comment by natasha-aleksandrova** [comment]
@ikempcode33 it appears that you have made changes to .gitignore and this cannot be merged as is.

please remove the changes

**Comment by ikempcode33** [comment]
HI, I do not see .gitignore in the directories and cannot find it. I  do not know how to remove the changes. Please help.

 

 

 

----------------------------------------------------

Isabella Kemp 

Department of Bioengineering

Undergraduate Researcher

University of Washington Class of 2021

Ikemp33@uw.edu

**Comment by ikempcode33** [comment]
I fixed the mailroom part 2 so you can exit in the loop/ However, I am not sure what is going on with the .gitignore, there a lot of files that are showing up for me if i don't ignore them, not sure what to do and why this is conflicting.

**Comment by natasha-aleksandrova** [comment]
@ikempcode33 within this PR, take a look at the Files Changed tab - you will see the changes to the .gitignore file. This might have been automatically or mistakely.. not sure. 

You will need to remove the changes otherwise this PR is not mergeable.

**Comment by natasha-aleksandrova** [comment]
@ikempcode33 if you intentionally made then change, then you can create .gitignore file in *in your own directory* but still will need to remove any changes from the root.

**Comment by ikempcode33** [comment]
Hi Natasha,

 

I pulled the changes back into my GitHub desktop, but now I have 50 files that say .json that are in my github desktop and I am not sure how to get rid of them so that they are no longer in here. Please help.

 

Isabella 

 

----------------------------------------------------

Isabella Kemp 

Department of Bioengineering

Undergraduate Researcher

University of Washington Class of 2021

Ikemp33@uw.edu

**Comment by ikempcode33** [comment]
I now have discarded all of the changes on my end in the Git Hub desktop and they seem to have gone away. Hopefully this resolved it! 

 

 

 

----------------------------------------------------

Isabella Kemp 

Department of Bioengineering

Undergraduate Researcher

University of Washington Class of 2021

Ikemp33@uw.edu

**Comment by natasha-aleksandrova** [comment]
@ikempcode33 looks much better, however you have removed a line from .gitignore - please take a look at the files changed tab in this PR.

**Comment by natasha-aleksandrova** [comment]
@ikempcode33 Hi how are you doing? Wanted to check in with you and see if there's anything we can help with? Thank you!

**Comment by ikempcode33** [comment]
Hi Natasha,

I’m doing well I am just very busy with work at my university so sorry if I am going slow in the course, learning python is very important to me. GitHub has also been a difficult transition for me as I don’t understand it very well, just enough to submit my work. I have also looked at the files changed tab in this PR and it looks like the .gitignore has just been deleted and not sure if it will let me bring anything back.

Thank you for checking in.

Isabella K.

**Comment by natasha-aleksandrova** [comment]
@ikempcode33 no worries - we all have busy lives and completely understand 🙂 

If you ever feel like you could use some one on one meeting to go over any questions you have about github or anything else - feel free to schedule some office hours by sending us an email. We'd be happy to meet with you!


As far as this PR, github no longer reports a conflict - it still shows changes to .gitignore (you would need to expand the diff to see it). But I believe it should be safe to merge this now.

**Pull request merged**